sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -257,8 +270,56 @@ void testAssociateNewPatientForHandleSameStateTwice() {
     @Test
     @DisplayName("associateNewLocations: Associate new location correctly")
     void testAssociateNewLocationForHandle() {
+        updated = false;
         associateNewLocationsSetup();
         final var expectedManipulationCalls = 2;
+        final List<MdibAccessObserver> observers = new ArrayList<>();
+
+        final MdibAccessObservable mockMdibAccessObservable = mock(MdibAccessObservable.class);
+        when(mockTestClient.getSdcRemoteDevice().getMdibAccessObservable()).thenReturn(mockMdibAccessObservable);
+        final AtomicReference<MdibAccessObserver> observer = new AtomicReference<>();
+        final Object signal = new Object();
+        doAnswer((arguments) -> {
+                    synchronized (signal) {
+                        observer.set(arguments.getArgument(0));
+                        signal.notifyAll();
+                    }
+                    return null;
+                })
+                .when(mockMdibAccessObservable)
+                .registerObserver(any());
+
+        final Thread t = new Thread(() -> {
+            ManipulationPreconditions.AssociateLocationsManipulation.EpisodicContextReportStateHandleObserver
+                    mdibAccessObserver;
+            for (int i = 0; i < 2; i++) {
+                mdibAccessObserver =
+                        (ManipulationPreconditions.AssociateLocationsManipulation
+                                        .EpisodicContextReportStateHandleObserver)
+                                observer.get();
+                synchronized (signal) {
+                    while (mdibAccessObserver == null) {
+                        try {
+                            signal.wait(300);
+                        } catch (InterruptedException e) {
```

Comment:
do not catch it here if you are not reacting to it

also you are not even interrupting the thread from another thread, thus there is no good way of reacting to it and it should just be escalated

Reply:
Found a solution together wth @maximilianpilz .

Reply:
Done.